src/panels/config/dashboard/ha-config-navigation.ts

```diff
@@ -19,10 +20,16 @@ class HaConfigNavigation extends LitElement {
 
   @property() public pages!: PageNavigation[];
 
+  @property() public externalConfig?: ExternalConfig;
+
   protected render(): TemplateResult {
     return html`
       ${this.pages.map((page) =>
-        canShowPage(this.hass, page)
+        (
+          page.path === "#external-app-configuration"
+            ? !this.externalConfig?.hasSettingsScreen
```

Comment:
```suggestion
            ? this.externalConfig?.hasSettingsScreen
```
Added for testing until we are set on a color for the entry